FreeRTOS/Test/CMock/smp/multiple_priorities_preemption_disable/multiple_priorities_preemption_disable_utest.c

```diff
@@ -179,3 +179,978 @@ int suiteTearDown( int numFailures )
+                                            EXIT_LOOP );
+
+
+    /* API Call */
+    portTASK_FUNCTION( prvIdleTask, args );
+
+    Try
+    {
+        prvIdleTask( NULL );
+    }
+    Catch( e )
+    {
+        if( e == EXIT_LOOP )
+        {
+            TEST_PASS();
+        }
+        else
+        {
+            TEST_FAIL();
+        }
+    }
+    /* Test Verifications */
+
+    /* this function (vPortSuppressTicksAndSleep_Expect) not being called is
+     * the aim of this test, it proves that the
+     * task  did not go to sleep, technically nothing happens */
+}
+
+/**
+ * @brief This test ensures that when we call prvCreateIdleTasks with and idle
+ *        name that is just as long as configMAX_TASK_NAME_LEN
+ *        no core id number is added at the end
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ * prvCreateIdleTasks();
+ *
+ * if( x < configMAX_TASK_NAME_LEN )
+ *
+ * @endcode
+ *
+ * configNMBER_OF_CORES > 1
+ */
+void test_prvCreateIdleTasks_name_too_long( void )
+{
+    BaseType_t prvCreateIdleTasks( void );
+
+    TCB_t * xIdleTask;
+    TCB_t xTask = { 0 };
+    int i;
+
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        pxCurrentTCBs[ i ] = &xTask;
+    }
+
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        /* prvInitialiseNewTask */
+        vListInitialiseItem_ExpectAnyArgs();
+        vListInitialiseItem_ExpectAnyArgs();
+        listSET_LIST_ITEM_VALUE_ExpectAnyArgs();
+        pxPortInitialiseStack_ExpectAnyArgsAndReturn( NULL );
+
+        vFakePortEnterCriticalSection_Expect();
+        listINSERT_END_ExpectAnyArgs();
+        portSetupTCB_CB_ExpectAnyArgs();
+        vFakePortGetCoreID_ExpectAndReturn( 0 );
+        vFakePortExitCriticalSection_Expect();
+    }
+
+    /* API Call */
+    prvCreateIdleTasks();
+
+    /* Test Verifications */
+    xIdleTask = ( TCB_t * ) xIdleTaskHandles[ 0 ];
+    TEST_ASSERT_EQUAL_STRING( configIDLE_TASK_NAME, xIdleTask->pcTaskName );
+}
+
+/**
+ * @brief This test ensures that if the scheduler is not running, and the
+ *        scheduler is suspended, taskSCHEDULER_SUSPENDED is returned
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ * xTaskGetSchedulerState();
+ *
+ * taskENTER_CRITICAL();
+ * taskEXIT_CRITICAL();
+ *
+ * @endcode
+ *
+ * configNMBER_OF_CORES > 1
+ * INCLUDE_xTaskGetSchedulerState = 1
+ * configUSE_TIMERS = 1
+ */
+void test_xTaskGetSchedulerState_scheduler_not_running_and_suspended( void )
+{
+    BaseType_t xRet;
+
+    xSchedulerRunning = pdTRUE;
+    uxSchedulerSuspended = pdTRUE;
+
+    vFakePortEnterCriticalSection_Expect();
+    vFakePortExitCriticalSection_Expect();
+
+    xRet = xTaskGetSchedulerState();
+
+    TEST_ASSERT_EQUAL( taskSCHEDULER_SUSPENDED, xRet );
+}
+
+/**
+ * @brief This test ensures that if the notify is zero and the ticks to wait are
+ *        greater than zero, the task is yielded
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *
+ * vTaskYieldWithinAPI();
```

Comment:
It is suggested that to include the branch condition which the test wants to verify in the code snippet.
```suggestion
 * if( xTicksToWait > ( TickType_t ) 0 )
 * {
 *         ...
 *         vTaskYieldWithinAPI();
 *         ...
 * }
 * else
 * {
 *     mtCOVERAGE_TEST_MARKER();
 * }
```